flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -794,7 +802,23 @@ public void goToWaitingForResources() {
                         LOG,
                         desiredResources,
                         this.initialResourceAllocationTimeout,
-                        this.resourceStabilizationTimeout));
+                        this.resourceStabilizationTimeout,
+                        null));
+    }
+
+    @Override
+    public void goToWaitingForResources(ExecutionGraph executionGraph) {
```

Comment:
I'm wondering, do we need an extra `goToWaitingForResources` method? 🤔 

Can we do something along these lines instead?

```
        final ExecutionGraph previousExecutionGraph = state.as(StateWithExecutionGraph.class)
                .map(StateWithExecutionGraph::getExecutionGraph)
                .orElse(null);
```

Reply:
I agree, `state.as` seems error-prone to me. Initial implementation had an error related to that (state was never the one expected).